libs/react-helpers/src/lib/generic-data-provider.ts

```diff
@@ -70,20 +119,60 @@ function makeDataProviderInternal<QueryData, Data, SubscriptionData, Delta>(
   let loading = false;
   let client: ApolloClient<object> | undefined = undefined;
   let subscription: Subscription | undefined = undefined;
+  let pageInfo: PageInfo | null = null;
+  let totalCount: number | undefined;
 
   // notify single callback about current state, delta is passes optionally only if notify was invoked onNext
-  const notify = (callback: UpdateCallback<Data, Delta>, delta?: Delta) => {
+  const notify = (
+    callback: UpdateCallback<Data, Delta>,
+    updateData?: { delta?: Delta; insertData?: Data | null }
```

Comment:
nit (non-blocking): `updateData` also sounds callable, `dataUpdate` on the other hand sounds more like a piece of data which is meant to update something with. okay, will try to stop being a variable-name-nazi, i think i've dropped enough comments on the topic :)

Reply:
Good point. I renamed it to `dataUpdate`